test/common/router/config_impl_test.cc

```diff
@@ -2029,5 +2034,229 @@ TEST(RoutePropertyTest, excludeVHRateLimits) {
   EXPECT_TRUE(config_ptr->route(headers, 0)->routeEntry()->includeVirtualHostRateLimits());
 }
 
+TEST(CustomRequestHeadersTest, AddNewHeader) {
```

Comment:
ok understood.. i will keep the existing test cases as is and add more for the req_header_formatter in a separate test file.